lib/dor/text_extraction/ocr.rb

```diff
@@ -89,6 +77,52 @@ def filenames_to_ocr
 
       private
 
+      # e.g. /abbyy/INPUT/ab123cd4567
+      def cleanup_input_folder
+        return unless Dir.exist?(abbyy_input_path)
+
+        logger.info "Removing empty ABBYY input directory: #{abbyy_input_path}"
+        FileUtils.rm_r(abbyy_input_path)
+      end
+
+      # e.g. /abbyy/OUTPUT/ab123cd4567
+      def cleanup_output_folder
+        return unless Dir.exist?(abbyy_output_path)
+
+        logger.info "Removing ABBYY output directory: #{abbyy_output_path}"
+        FileUtils.rm_r(abbyy_output_path)
+      end
+
+      # e.g. /abbyy/INPUT/ab123cd4567.xml
+      def cleanup_xml_ticket
+        xml_ticket_file = Abbyy::Ticket.new(filepaths: [], druid: cocina_object.externalIdentifier).file_path
+
+        return unless File.exist?(xml_ticket_file)
+
+        logger.info "Removing XML Ticket File: #{xml_ticket_file}"
+        FileUtils.rm_r(xml_ticket_file)
+      end
+
+      # e.g. /abbyy/RESULTXML/ab123cd4567.xml.result.xml
+      def cleanup_abbyy_results
+        abbyy_results = Dor::TextExtraction::Abbyy::Results.find_latest(druid: bare_druid, logger:)
+
+        # this could be nil if there is no latest result XML file for this druid
+        return unless abbyy_results && File.exist?(abbyy_results.result_path)
+
+        logger.info "Removing XML Result File: #{abbyy_results.result_path}"
+        FileUtils.rm_r(abbyy_results.result_path)
+      end
+
+      # e.g. /abbyy/EXCEPTIONS/druid:ab123cd4567.xml.result.xml
+      def cleanup_abbyy_exceptions
+        abbyy_exceptions = Dir.glob("#{Settings.sdr.abbyy.local_exception_path}/*#{bare_druid}*.xml")
+        abbyy_exceptions.each do |abbyy_exception_file|
+          logger.info "Removing XML Exception File: #{abbyy_exception_file}"
+          FileUtils.rm_r(abbyy_exception_file)
+        end
+      end
```

Comment:
move all of various cleanup bits into their own methods to make the `cleanup` method itself easier to reason with and easily retriable

also guard against the files/folders not existing ... will just be a noop